bodo/numba_compat.py

```diff
@@ -2975,6 +3005,9 @@ def CallConstraint_resolve(self, typeinfer, typevars, fnty):
     for a in itertools.chain(pos_args, kw_args.values()):
         # Forbids imprecise type except array of undefined dtype
         if not a.is_precise() and not isinstance(a, types.Array):
+            # Bodo change: allow streaming state type to be imprecise
```

Comment:
For sustainability it might be safer to wrap this in an API or a property of the type. Not super critical right now, but this will make this easier to extend in anything for streaming Python requires it.